### PR TITLE
Day twenteen

### PR DESCRIPTION
**Pull request by Talaspaev**

### CODE REVIEW

File: src/Components/Users/users.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+import { I18Y } from '../../constants';
+
+import style from './users.module.css';
+
+const Users = (props) => {
+  if (props.users.length === 0) {
+    props.setUsers([{
```
Comment (Dmitry-White) [inline src/Components/Users/users.jsx:9]: Consider moving this array of objects to a separate variable outsider of Users.

File: src/Components/Users/users.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      id: 3, followed: false, fullname: 'Natalya', status: 'consectetur ', photoUrl: 'https://images11.cosmopolitan.ru/upload/img_cache/7b3/7b3b3da698f08bee6e9e7402053f6df5_ce_750x497x0x129_fitted_740x0.jpg', location: { City: 'Minsk', Country: 'Belarus' },
+    },
+    {
+      id: 4, followed: true, fullname: 'Igor', status: 'architecto ', photoUrl: 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRNtwQUeVJ4vImAb6HiD2qFiEXchzJJKPDWUU96D33THFtBskw1', location: { City: 'Minsk', Country: 'Belarus' },
+    },
+    {
+      id: 5, followed: true, fullname: 'Adil', status: 'adipisicing elit', photoUrl: 'https://img.gazeta.ru/files3/784/11560784/skrskr2-pic905-895x505-4108.jpg', location: { City: 'Minsk', Country: 'Belarus' },
+    }]);
+  }
+  const usersItem = props.users.map((item) => (
```
Comment (Dmitry-White) [inline src/Components/Users/users.jsx:25]: `usersItem` is essentially not a single thing, so consider creating separate functions for `usersItem` and `usersItems`.

File: src/Components/Users/users.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    <div key={item.id} className={style.item}>
+
+      <div>
+        <img src={item.photoUrl} alt="" className={style.img} />
+        {item.followed ? <button type="button" onClick={() => { props.unfollowUser(item.id); }}>{I18Y.fl}</button>
+          : <button type="button" onClick={() => { props.followUser(item.id); }}>{I18Y.unfl}</button>}
```
Comment (Dmitry-White) [inline src/Components/Users/users.jsx:31]: Consider moving click handlers into separate functions.

File: src/Components/Users/users.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    {
+      id: 5, followed: true, fullname: 'Adil', status: 'adipisicing elit', photoUrl: 'https://img.gazeta.ru/files3/784/11560784/skrskr2-pic905-895x505-4108.jpg', location: { City: 'Minsk', Country: 'Belarus' },
+    }]);
+  }
+  const usersItem = props.users.map((item) => (
+    <div key={item.id} className={style.item}>
+
+      <div>
+        <img src={item.photoUrl} alt="" className={style.img} />
+        {item.followed ? <button type="button" onClick={() => { props.unfollowUser(item.id); }}>{I18Y.fl}</button>
```
Comment (Dmitry-White) [inline src/Components/Users/users.jsx:30]: Consider moving this follow/unfollow logic into a separate function.

File: src/constants.js [hunk 5]
```diff
@@ -3,6 +3,8 @@ const I18Y = {
   qus: 'How are you?',
   ans: 'I am exhausted',
   btn: 'Button',
+  fl: 'follow',
```
Comment (Dmitry-White) [inline src/constants.js:6]: Consider restructuring this I18N object to have 2 separate fields "RU" and "ENG" insider which are objects with all the keys and their translated values inside.
`const I18N = { ENG: { ..., FOLLOW: 'Follow', ....}, RU: {..., FOLLOW: 'Подписаться', ... } };`

File: src/redux/redux-state.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { createStore, combineReducers } from 'redux';
+
+import profileReducer from './profileReducer';
+import dialogsReducer from './dialogsReducer';
+import usersReducer from './usersReducer';
+
+
```
Comment (Dmitry-White) [inline src/redux/redux-state.js:7]: Consider removing this redundant empty line.

File: src/Components/Dialogs/dialogsContainer.jsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { connect } from 'react-redux';
+import { addMessageActionCreator, updateMessageActionCreator } from '../../redux/redux-state';
+
+import Dialogs from './dialogs';
+
+const mapStateToProps = (state) => ({
+  dialogsPages: state.dialogsPages,
+});
+const mapDispatchToProps = (dispatch) => ({
+  AddNewMessage: () => {
```
Comment (Dmitry-White) [inline src/Components/Dialogs/dialogsContainer.jsx:10]: Consider renaming such methods to have a lowercase letter at the start of the name.

File: src/Components/Users/users.jsx (hunk 1 above)
Comment (Talaspaev) [reply src/Components/Users/users.jsx:9]: i did it.

File: src/Components/Users/users.jsx (hunk 2 above)
Comment (Talaspaev) [reply src/Components/Users/users.jsx:25]: sure.

File: src/Components/Users/users.jsx (hunk 3 above)
Comment (Talaspaev) [reply src/Components/Users/users.jsx:31]: i will  move it next PR

File: src/Components/Users/users.jsx (hunk 4 above)
Comment (Talaspaev) [reply src/Components/Users/users.jsx:30]: i will move it next PR

File: src/constants.js (hunk 5 above)
Comment (Talaspaev) [reply src/constants.js:6]: i added new object

File: src/redux/redux-state.js (hunk 6 above)
Comment (Talaspaev) [reply src/redux/redux-state.js:7]: sure it's redundant line

File: src/Components/Dialogs/dialogsContainer.jsx (hunk 7 above)
Comment (Talaspaev) [reply src/Components/Dialogs/dialogsContainer.jsx:10]: sure

**Pull request closed without merging**